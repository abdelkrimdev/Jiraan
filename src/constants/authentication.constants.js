export const authenticationConstants = {
    // User Sign Up
    USER_CREATE_REQUEST: 'FETCH_CREATE_USER_REQUEST',
    USER_CREATE_SUCCESS: 'FETCH_CREATE_USER_SUCCESS',
    USER_CREATE_FAILURE: 'FETCH_CREATE_USER_FAILURE',
    // User Sign In
    USER_LOGIN_REQUEST: 'FETCH_LOGIN_USER_REQUEST',
    USER_LOGIN_SUCCESS: 'FETCH_LOGIN_USER_SUCCESS',
    USER_LOGIN_FAILURE: 'FETCH_LOGIN_USER_FAILURE',
    // Get Current User
    USER_CURRENT_ACTION: 'FIRE_CURRENT_USER_ACTION',
    // User Sign Out
    USER_LOGOUT_ACTION: 'FIRE_LOGOUT_USER_ACTION'
};
