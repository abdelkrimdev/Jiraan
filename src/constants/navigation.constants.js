export const navigationConstants = {
  // Switch Navigator Keys
  SPLASH: 'Splash',
  AUTH: 'Auth',
  APP: 'App',
  // Authentication Stack Keys
  LOGIN: 'Login'
}
