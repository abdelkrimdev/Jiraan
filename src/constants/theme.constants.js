export const themeConstants = {
  // Theme Colors
  PRIMARY_COLOR: '#8e44ad',
  SECONDARY_COLOR: '#95a5a6',
  SUCCESS_COLOR: '#27ae60',
  DANGER_COLOR: '#e74c3c',
  WARNING_COLOR: '#f1c40f',
  INFO_COLOR: '#3498db',
  LIGHT_COLOR: '#ecf0f1',
  DARK_COLOR: '#2c3e50'
}
