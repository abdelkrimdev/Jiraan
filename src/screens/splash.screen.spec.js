import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import ConnectedSplash, { Splash } from './splash.screen'
import { navigationConstants } from '../constants/navigation.constants'

const mockStore = configureStore([ ])
const navigation = { navigate: jest.fn() }

describe('splash screen', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('should render without crashing.', () => {
    const wrapper = shallow(<Splash />)

    expect(wrapper).toBeDefined()
  })

  it('should render correctly.', () => {
    const wrapper = shallow(<Splash />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should connect to the store without crashing.', () => {
    const store = mockStore({
      currentUser: { user: { } }
    })

    const container = shallow(<ConnectedSplash store={store} />).dive()

    expect(container).toBeDefined()
  })

  it('should redirect to authentication stack when the user is not logged in.', () => {
    const wrapper = shallow(<Splash navigation={navigation} />)
    wrapper.setProps({ authenticatedUser: null })

    expect(navigation.navigate).toHaveBeenCalled()
    expect(navigation.navigate).toHaveBeenCalledWith(navigationConstants.AUTH)
  })

  it('should redirect to application stack when the user is logged in.', () => {
    const wrapper = shallow(<Splash navigation={navigation} />)
    wrapper.setProps({ authenticatedUser: { } })

    expect(navigation.navigate).toHaveBeenCalled()
    expect(navigation.navigate).toHaveBeenCalledWith(navigationConstants.APP)
  })
})
