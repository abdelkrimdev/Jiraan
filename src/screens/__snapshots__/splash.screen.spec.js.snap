// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splash screen should dispatch the getCurrentUser action when component is mounting. 1`] = `
Array [
  Object {
    "type": "FIRE_GET_CURRENT_USER_ACTION",
    "user": null,
  },
]
`;

exports[`splash screen should render correctly. 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
  <StatusBar
    animated={false}
    barStyle="default"
    showHideTransition="fade"
  />
</Component>
`;
